src/pyodide-py/pyodide/_base.py

```diff
@@ -124,9 +180,18 @@ def _split_and_compile(self, code: str, flags: int = 0x0) -> Tuple[Any, Any]:
         if not mod.body:
             return None, None
 
+        mode = self.mode
+        if mode == "last_expr_or_assign":
+            self._last_assign_to_expr(mod)
+            mode = "last_expr"
```

Comment:
```suggestion
        if self.mode == "last_expr_or_assign":
            # If the last statement is an assignment, add an extra expression
            # to the end with just the L-value so that we can handle it with 
            # the last_expr code.
            self._last_assign_to_expr(mod)
```